spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java

```diff
@@ -0,0 +1,139 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+public class SCServerAuthMethod implements ISCAuth {
+
+    private static String LOG_TAG = SCServerAuthMethod.class.getSimpleName();
+    private static final String JSON_WEB_TOKEN = "x-access-token";
+    private static final String USERNAME = "username";
+    private static final String PWD = "pwd";
+    private Context context;
+    private String jsonWebToken;
+    private SecureSharedPreferences settings;
+    private String serverUrl;
+
+
+    public SCServerAuthMethod(Context context, String serverUrl ) {
+        this.context = context;
+        this.settings = new SecureSharedPreferences(context);
+        this.serverUrl = serverUrl;
+    }
+
+    @Override
+    public boolean authFromCache() {
+        String u = getUsername();
+        String p = getPassword();
+        if (u != null && p != null) {
+            return authenticate(u,p);
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean authenticate(String username, String pwd) {
+        return auth(username, pwd);
+    }
+
+    @Override
+    public void logout() {
+        removeCredentials();
+    }
+
+    @Override
+    public String xAccessToken() {
+        return settings.getString(JSON_WEB_TOKEN, null);
+    }
+
+    @Override
+    public String username() {
+        return getUsername();
+    }
+
+    private boolean auth(final String username, final String pwd) {
+        boolean authed = false;
+        final String theUrl = String.format(Locale.US, "%s/api/authenticate", serverUrl);
+        Response response = HttpHandler.getInstance()
+                .post(theUrl, String.format("{\"email\": \"%s\", \"password\":\"%s\"}", username, pwd))
+                .toBlocking()
+                .first();
+
+        if (response.isSuccessful()) {
+            try {
+                jsonWebToken = new JSONObject(response.body().string())
+                        .getJSONObject("result").getString("token");
+                if (jsonWebToken != null) {
+                    saveAccessToken(jsonWebToken);
+                    saveCredentials(username, pwd);
+                    authed = true;
+                }
+            } catch (JSONException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            authed = false;
```

Comment:
Won't `authed` always be false anytime it gets to this line?  If so, we can probably remove it b/c it's redundant